### PR TITLE
Update JDK test matrix to currently supported versions (8, 11, 13)

### PR DESCRIPTION
**Pull request by fbenz**
JDK 8 and 11 are long-term support (LTS) versions.

JDK 13 is supported until JDK 14 is released in March 2020.

Further steps to get everything working on JDK 13:
* Update Maven plugins.
* Update JAXB dependencies.
* Updated example projects to Spring Boot 2.2.4. Spring Boot 2.2 (Spring Framework 5.2) added JDK 13 support.
* Update of Gradle.

### CODE REVIEW

**Comment by jmisur** [comment]
What about jdk 12? Also build 13 needs some tweaking probably.

**Comment by fbenz** [comment]
JDK 12 is not supported anymore. I need to adjust several POMs because of the explicitly JDK versions in there. Actually, I don't expect any changes for JDK 13 but we will see when I try it out.

Additional planned updates from my side:
* Update example projects to the latest Spring Boot version. The main adjustment is due to changes in Spring HATEOS.
* Update to the latest Dokka version. There are issues with the copied over test code and I'm trying to build a test JAR. This should make future updates easier.

**Comment by fbenz** [comment]
The sample projects need to be upgraded to Spring Boot 2.2 to support Java 13: https://spring.io/blog/2019/10/16/spring-boot-2-2-0
Currently, it fails with: ASM ClassReader failed to parse class file - probably due to a new Java class file version that isn't supported yet

File: samples/java-webmvc/generated-docs/index.html [hunk 1]
```diff
@@ -2337,30 +2337,9 @@ <h4 id="_query_parameters_7"><a class="anchor" href="#_query_parameters_7"></a><
...
-<tr>
-<td class="tableblock halign-left valign-top"><p class="tableblock">command</p></td>
-<td class="tableblock halign-left valign-top"><p class="tableblock">String</p></td>
-<td class="tableblock halign-left valign-top"><p class="tableblock">false</p></td>
-<td class="tableblock halign-left valign-top"><p class="tableblock">Command to execute.</p></td>
-</tr>
-</tbody>
-</table>
+<div class="paragraph">
+<p>No request body.</p>
```
Comment (jmisur) [inline samples/java-webmvc/generated-docs/index.html]: Something changed here.

File: samples/kotlin-webmvc/generated-docs/index.html [hunk 2]
```diff
@@ -2195,30 +2195,9 @@ <h4 id="_query_parameters_7"><a class="anchor" href="#_query_parameters_7"></a><
...
-<tr>
-<td class="tableblock halign-left valign-top"><p class="tableblock">command</p></td>
-<td class="tableblock halign-left valign-top"><p class="tableblock">String</p></td>
-<td class="tableblock halign-left valign-top"><p class="tableblock">false</p></td>
-<td class="tableblock halign-left valign-top"><p class="tableblock">Command to execute.</p></td>
-</tr>
-</tbody>
-</table>
+<div class="paragraph">
+<p>No request body.</p>
```
Comment (jmisur) [inline samples/kotlin-webmvc/generated-docs/index.html]: Also here.

File: pom.xml [hunk 3]
```diff
@@ -193,32 +193,14 @@
             </modules>
         </profile>
         <profile>
-            <id>java9</id>
```
[the remark this answers is not in the document]
Comment (fbenz) [reply pom.xml]: I still think that it's the wrong signal and not worth any additional effort. If Java 8 & 11 work, it's unlikely to be broken for Java 9. Spring Boot is tested against 8, 11 and 13: https://github.com/spring-projects/spring-boot/blob/master/ci/pipeline.yml#L169

File: samples/java-webmvc/generated-docs/index.html (hunk 1 above)
Comment (fbenz) [reply samples/java-webmvc/generated-docs/index.html]: Fixed by adjusting the `MockMvcBase` to the new ModelAttribute support.

File: pom.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (fbenz) [reply pom.xml]: If you still insist, I would propose a follow PR from your side 😉

File: samples/java-webmvc/generated-docs/index.html [hunk 4]
```diff
@@ -1143,6 +1143,46 @@ <h4 id="_request_fields_3"><a class="anchor" href="#_request_fields_3"></a><a cl
...
+</tr>
+</thead>
+<tbody>
+<tr>
+<td class="tableblock halign-left valign-top"><p class="tableblock">description</p></td>
+<td class="tableblock halign-left valign-top"><p class="tableblock">String</p></td>
+<td class="tableblock halign-left valign-top"><p class="tableblock">true<br>
+EN: false</p></td>
+<td class="tableblock halign-left valign-top"><p class="tableblock">Some information about the item.</p>
+<p class="tableblock">DE: Size must be between 2 and 10 inclusive.<br>
```
Comment (jmisur) [inline samples/java-webmvc/generated-docs/index.html]: This is a new block

File: samples/java-webmvc/generated-docs/index.html [hunk 5]
```diff
@@ -1269,7 +1309,47 @@ <h4 id="_query_parameters_2"><a class="anchor" href="#_query_parameters_2"></a><
...
+EN: Size must be between 4 and 12 inclusive.<br>
```
Comment (jmisur) [inline samples/java-webmvc/generated-docs/index.html]: same

File: samples/java-webmvc/generated-docs/index.html (hunk 4 above)
Comment (fbenz) [reply samples/java-webmvc/generated-docs/index.html]: This block is also on master https://github.com/ScaCap/spring-auto-restdocs/blob/master/samples/java-webmvc/generated-docs/index.html#L1146 so it's strange that appears in the diff.

**Pull request merged**